### ISSUE TITLE
`ImplicitDefaultLocale` should also catch `"%d".format(1)`

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#6342]
## Expected Behavior of the rule
The rule `ImplicitDefaultLocale` catches usages of `String.format` without a locale. Using the format extension function without a locale should also be non-compliant.

```kotlin
// non-compliant
String.format("%d", 1)
"%d".format(1)

// compliant
String.format("%d", Locale.US, 1)
"%d".format(Locale.US, 1)
```

**Comment by atulgpt** [comment]
Hi, @marschwar I think for handling the`"%d".format(1)` case we may need to make this rule dependent on TR what are your thoughts on this?

**Comment by marschwar** [comment]
It already depends on type resolution. It is one of those rules with mixed behavior.
See #6342
